src/binder/assemblybinder.cpp

```diff
@@ -685,19 +685,23 @@ namespace BINDER_SPACE
 
         StackSString sCoreLib;
 
-        // At run-time, System.Private.CoreLib.ni.dll is typically always available, and
-        // System.Private.CoreLib.dll is typically not.  So check for the NI first.
+        // At run-time, System.Private.CoreLib.dll is expected to be the NI image.
+        // Try binding to it first as that. If it fails, we will try to bind to it as 
+        // IL image.
         sCoreLib = sCoreLibDir;
-        sCoreLib.Append(CoreLibName_NI_W);
+        sCoreLib.Append(CoreLibName_IL_W);
+#if defined(FEATURE_PAL) && !defined(_TARGET_AMD64_)
+        // Native image for core library only exists for X64 on non-Windows. As other architectures
+        // bring them up, we should update this condition.
+        fBindToNativeImage = false;
```

Comment:
> I dont have a way to setup a repro for running Linux Arm tests

@gkhanna79, do you mean that you don't have a Linux ARM device for testing or is the problem not reproing always?